### PR TITLE
Replace k8s client-go mocks with fakes

### PR DESCRIPTION
**Pull request by bbarnes52**
This revision removes the generated mocks for the k8s client-go package from the spire repo. Instead, the various `fake` packages from the client-go repository (the canonical method for stubbing out k8s client calls in unit tests) are used.

This has a few benefits:
1. The tests themselves are cleaner. From https://testing.googleblog.com/2013/05/testing-on-toilet-dont-overuse-mocks.html, "Some signs that you're overusing mocks are if you're mocking out more than one or two classes, or if one of your mocks specifies how more than one or two methods should behave."

2. The k8s client-go repository does a poor job adhering to Go module management best practices, backwards incompatible changes across minor versions are commonplace. See https://github.com/kubernetes/client-go/issues/234. By generating mocks for entire packages in the client-go repo, any backwards incompatible changes to the interface of these packages cause the spire repo to be incompatible with the newer version of the `k8s.io/client_go` module (even if spire does not actually use the methods in question). This makes it challenging to depend on the spire repo, as if any other modules in the dependency chain use the k8s client and request a minimum version of `k8s.io/client-go` higher than what spire requests (currently `0.18.2`) it is not uncommon for builds to fail due to backwards incompatible changes that affect the generated mocks. For example, here is the error message when we attempted to upgrade the version of `k8s.io/client-go` in our repo:
```
go: finding module for package k8s.io/client-go/kubernetes/typed/settings/v1alpha1
go: finding module for package k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1
go: downloading k8s.io/mount-utils v0.20.1
go: downloading k8s.io/component-helpers v0.20.1
go: downloading k8s.io/controller-manager v0.20.1
code.uber.internal imports
	github.com/spiffe/spire/cmd/spire-server imports
	github.com/spiffe/spire/cmd/spire-server/cli imports
	github.com/spiffe/spire/cmd/spire-server/cli/run imports
	github.com/spiffe/spire/pkg/server imports
	github.com/spiffe/spire/pkg/server/catalog imports
	github.com/spiffe/spire/pkg/server/plugin/nodeattestor/k8s/psat imports
	github.com/spiffe/spire/pkg/common/plugin/k8s/apiserver tested by
	github.com/spiffe/spire/pkg/common/plugin/k8s/apiserver.test imports
	github.com/spiffe/spire/test/mock/common/plugin/k8s/clientset imports
	k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1: module k8s.io/client-go@latest found (v1.5.2, replaced by k8s.io/client-go@v0.20.1), but does not contain package k8s.io/client-go/kubernetes/typed/auditregistration/v1alpha1
code.uber.internal imports
	github.com/spiffe/spire/cmd/spire-server imports
	github.com/spiffe/spire/cmd/spire-server/cli imports
	github.com/spiffe/spire/cmd/spire-server/cli/run imports
	github.com/spiffe/spire/pkg/server imports
	github.com/spiffe/spire/pkg/server/catalog imports
	github.com/spiffe/spire/pkg/server/plugin/nodeattestor/k8s/psat imports
	github.com/spiffe/spire/pkg/common/plugin/k8s/apiserver tested by
	github.com/spiffe/spire/pkg/common/plugin/k8s/apiserver.test imports
	github.com/spiffe/spire/test/mock/common/plugin/k8s/clientset imports
k8s.io/client-go/kubernetes/typed/settings/v1alpha1: module k8s.io/client-go@latest found (v1.5.2, replaced by k8s.io/client-go@v0.20.1), but does not contain package k8s.io/client-go/kubernetes/typed/settings/v1alpha1
```

While the root cause here is k8s poor Go module management, we can effectively mitigate this problem by limiting our exposure to the client-go API surface to just the methods we actually use.

Signed-off-by: Brian Barnes <bgb@uber.com>

### CODE REVIEW

**Comment by azdagron** [comment]
Huzzah! Any change that removes mock use in favor of fakes has my support and adoration :) We're a little heads down with preparations for 1.0, but I'll review this as soon as I get a minute.

**Review by rturner3 — APPROVED** [review]
LGTM!

File: pkg/common/plugin/k8s/apiserver/client_test.go [hunk 1]
```diff
@@ -117,7 +114,7 @@ oIrPuyjOmscrC627wX3LGUHwPKtNArBT8lKFfda1B1BqAk0q1/ui/A==
 )
 
 const (
-	testToken = "TEST-TOKEN"
+	_testToken = "TEST-TOKEN"
```
Comment (azdagron) [inline pkg/common/plugin/k8s/apiserver/client_test.go:117]: nit: SPIRE doesn't use this convention for unexported package-level vars

**Review by azdagron — DISMISSED** [review]
\o/, thanks @bbarnes52

File: pkg/common/plugin/k8s/apiserver/client_test.go (hunk 1 above)
Comment (zmt) [reply pkg/common/plugin/k8s/apiserver/client_test.go]: Perhaps we should reference a specific style guide in the CONTRIBUTING.md. This convention comes directly from https://github.com/uber-go/guide/blob/master/style.md#prefix-unexported-globals-with-_

**Comment by zmt** [comment]
Great analysis and thanks for the fix. I'm looking forward to getting this on an upgrade soon. minimizing and untangling transitive dependency conflicts w/ k8s.io components is a big win.

**Review by rturner3 — APPROVED** [review]

File: pkg/common/plugin/k8s/apiserver/client_test.go (hunk 1 above)
Comment (azdagron) [reply pkg/common/plugin/k8s/apiserver/client_test.go]: A style guide has been something we've talked about and wanted to introduce before, it has just never been prioritized. I've opened an issue to track this (#2377). Thanks for bringing it up.

**Pull request merged**